### PR TITLE
Fixed histogram substraction bug with NEON instructions

### PR DESCRIPTION
**Pull request by mshabunin**
Same as PR #5521 without clutter

### CODE REVIEW

**Comment by mshabunin** [comment]
@jet47 , is it OK?

File: modules/imgproc/src/smooth.cpp [hunk 1]
```diff
@@ -1845,14 +1845,14 @@ static inline void histogram_sub_simd( const HT x[16], HT y[16] )
 
 static inline void histogram_add_simd( const HT x[16], HT y[16] )
 {
-    vst1q_u16(y, vaddq_u16(vld1q_u16(x), vld1q_u16(y)));
-    vst1q_u16(y + 8, vaddq_u16(vld1q_u16(x + 8), vld1q_u16(y + 8)));
+    vst1q_u16(y, vaddq_u16(vld1q_u16(y), vld1q_u16(x)));
+    vst1q_u16(y + 8, vaddq_u16(vld1q_u16(y + 8), vld1q_u16(x + 8)));
```
Comment (ilya-lavrenov) [inline modules/imgproc/src/smooth.cpp]: what is the difference? a + b == b + a, not?

Comment (mshabunin) [inline modules/imgproc/src/smooth.cpp]: I guess not. Reverted.

**Comment by jet47** [comment]
:+1:

**Pull request merged**